pinot-query-planner/src/main/java/org/apache/pinot/query/planner/StagePlanner.java

```diff
@@ -95,12 +97,15 @@ private StageNode walkRelPlan(RelNode node, int currentStageId) {
     if (isExchangeNode(node)) {
       // 1. exchangeNode always have only one input, get its input converted as a new stage root.
       StageNode nextStageRoot = walkRelPlan(node.getInput(0), getNewStageId());
-      RelDistribution.Type exchangeType = ((LogicalExchange) node).distribution.getType();
+      RelDistribution distribution = ((LogicalExchange) node).getDistribution();
+      RelDistribution.Type exchangeType = distribution.getType();
 
       // 2. make an exchange sender and receiver node pair
-      StageNode mailboxReceiver = new MailboxReceiveNode(currentStageId, nextStageRoot.getStageId(), exchangeType);
-      StageNode mailboxSender = new MailboxSendNode(nextStageRoot.getStageId(), mailboxReceiver.getStageId(),
+      StageNode mailboxReceiver = new MailboxReceiveNode(currentStageId, node.getRowType(), nextStageRoot.getStageId(),
           exchangeType);
+      StageNode mailboxSender = new MailboxSendNode(nextStageRoot.getStageId(), node.getRowType(),
+          mailboxReceiver.getStageId(), exchangeType, exchangeType == RelDistribution.Type.HASH_DISTRIBUTED
+          ? new FieldSelectionKeySelector(distribution.getKeys().get(0)) : null);
```

Comment:
I thought initially we are going to support only broadcast based distribution / exchange ?

Reply:
yes. ideally, we should take `SqlHints` to decide whether we want to use broadcast or distributed hash sender. i can default enable the broadcast or hash. and keep the other method conditional

Reply:
yes totally agree on making this decision dictated by statistics / optimizer etc. My point was more on if we are hardcoding the exchange type in the initial implementation, it can be broadcast as I feel it is simpler to debug / get right initially may be